MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs

```diff
@@ -2076,7 +2075,77 @@ public void Mesh2dMakeUniformThroughAPI()
         }
 
         [Test]
-        public void Mesh2dMakeUniformOnExtensionThroughAPI()
+        public void Mesh2dMakeRectangularMeshFromPolygonThroughAPIFails()
+        {
+            // Setup
+            using (var api = new MeshKernelApi())
+            using (var polygon = new DisposableGeometryList())
+            {
+                var id = 0;
+                var mesh2d = new DisposableMesh2D();
+                try
+                {
+                    id = api.AllocateState(0);
+
+                    var makeGridParameters = MakeGridParameters.CreateDefault();
+
+                    makeGridParameters.GridAngle = 0.0;
+                    makeGridParameters.XGridBlockSize = 1.0;
+                    makeGridParameters.YGridBlockSize = 1.0;
+
+                    // geometry list is empty, expect an algorithm error to be thrown in the backend
+                    var algorithmErrorExitCode = -1;
+                    api.GetExitCodeAlgorithmError(ref algorithmErrorExitCode);
+                    Assert.AreEqual(algorithmErrorExitCode,api.Mesh2dMakeRectangularMeshFromPolygon(id, makeGridParameters, polygon));
+                }
+                finally
+                {
+                    api.DeallocateState(id);
+                    mesh2d.Dispose();
+                }
+            }
+        }
+
+
+        [Test]
+        public void Mesh2dMakeRectangularMeshFromPolygonThroughAPI()
+        {
+            // Setup
+            using (var api = new MeshKernelApi())
+            using (var polygon = new DisposableGeometryList())
+            {
+                var id = 0;
+                var mesh2d = new DisposableMesh2D();
+                try
+                {
+                    id = api.AllocateState(0);
+
+                    var makeGridParameters = MakeGridParameters.CreateDefault();
+
+                    makeGridParameters.GridAngle = 0.0;
+                    makeGridParameters.XGridBlockSize = 1.0;
+                    makeGridParameters.YGridBlockSize = 1.0;
+
+                    polygon.XCoordinates = new[] { 0.5, 2.5, 5.5, 3.5, 0.5 };
+                    polygon.YCoordinates = new[] { 2.5, 0.5, 3.0, 5.0, 2.5 };
+                    polygon.Values = new[] { 0.0, 0.0, 0.0, 0.0, 0.0 };
+                    polygon.NumberOfCoordinates = 5;
+
+                    Assert.AreEqual(0, api.Mesh2dMakeRectangularMeshFromPolygon(id, makeGridParameters, polygon));
+                    Assert.AreEqual(0, api.Mesh2dGetData(id, out mesh2d));

```

Comment:
you could also do it 

Assert.AreEqual(0, api.Mesh2dMakeRectangularMeshFromPolygon(id, makeGridParameters, polygon));
Assert.AreEqual(0, api.Mesh2dGetData(id, out DisposableMesh2D mesh2d));
Assert.NotNull(mesh2d);
Assert.AreEqual(9, mesh2d.NumNodes);
mesh2d.Dispose();

eliminating var mesh2d = new DisposableMesh2D(); above